ctapipe/tools/stage1.py

```diff
@@ -1,240 +1,81 @@
-    ).tag(config=True)
-
-    split_datasets_by = CaselessStrEnum(
-        values=["tel_id", "tel_type"],
-        default_value="tel_id",
-        help="Splitting level for the parameters and images datasets",
-    ).tag(config=True)
-
-    compression_type = CaselessStrEnum(
-        values=["blosc:zstd", "zlib"],
-        help="compressor algorithm to use. ",
-        default_value="blosc:zstd",
-    ).tag(config=True)
-
-    image_cleaner_type = create_class_enum_trait(
-        base_class=ImageCleaner, default_value="TailcutsImageCleaner"
-    )
-
-    write_index_tables = Bool(
-        help=(
-            "Generate PyTables index datasets for all tables that contain an "
-            "event_id or tel_id. These speed up in-kernal pytables operations,"
-            "but add some overhead to the file. They can also be generated "
-            "and attached after the file is written "
-        ),
-        default_value=False,
-    ).tag(config=True)
-
-    overwrite = Bool(help="overwrite output file if it exists").tag(config=True)
     progress_bar = Bool(help="show progress bar during processing").tag(config=True)
 
     aliases = {
         "input": "EventSource.input_url",
-        "output": "Stage1ProcessorTool.output_path",
+        "output": "DL1Writer.output_path",
         "allowed-tels": "EventSource.allowed_tels",
         "max-events": "EventSource.max_events",
-        "image-cleaner-type": "Stage1ProcessorTool.image_cleaner_type",
+        "image-cleaner-type": "ImageProcessor.image_cleaner_type",
     }
 
     flags = {
         "write-images": (
-            {"Stage1ProcessorTool": {"write_images": True}},
+            {"DL1Writer": {"write_images": True}},
             "store DL1/Event/Telescope images in output",
         ),
         "write-parameters": (
-            {"Stage1ProcessorTool": {"write_parameters": True}},
+            {"DL1Writer": {"write_parameters": True}},
             "store DL1/Event/Telescope parameters in output",
         ),
         "write-index-tables": (
-            {"Stage1ProcessorTool": {"write_index_tables": True}},
+            {"DL1Writer": {"write_index_tables": True}},
             "generate PyTables index tables for the parameter and image datasets",
         ),
         "overwrite": (
-            {"Stage1ProcessorTool": {"overwrite": True}},
+            {"DL1Writer": {"overwrite": True}},
             "Overwrite output file if it exists",
         ),
         "progress": (
-            {"Stage1ProcessorTool": {"progress_bar": True}},
+            {"Stage1Tool": {"progress_bar": True}},
             "show a progress bar during event processing",
         ),
     }
 
     classes = List(
-        [CameraCalibrator, ImageQualityQuery, EventSource]
+        [CameraCalibrator, EventSource]
         + classes_with_traits(ImageCleaner)
         + classes_with_traits(ImageExtractor)
         + classes_with_traits(GainSelector)
+        + classes_with_traits(DL1Writer)
```

Comment:
Isn't `classes_with_traits` though to be for abstract base classes? I think you should just add `DL1Writer` and `ImageProcessor` to the list without the `classes_with_traits` call. But I'm not sure.

Reply:
Yes, I think you are right - I had changed that at some point, but reverted it.  `classes_with_traits` (confusingly named), really just looks for child classes of the parent class.   So far DL1Writer has no sub-classes, so this doesn't really do anything other than include the parent (so this still works, but is probably redundant) .

Reply:
I think that means it only makes sense to use this where we use the `create_class_enum` approach, right? So for anything where we don't do that, it might actually be misleading, since there is no way the listed configuration options for subclasses actully take effect, since there is no way to make the tool use a subclass.

I would remove it then.

Reply:
ok, will remove